packages/sprotty/src/base/commands/command-stack.ts

```diff
@@ -210,6 +215,16 @@ export class CommandStack implements ICommandStack {
     protected handleCommand(command: ICommand,
                             operation: (context: CommandExecutionContext) => CommandReturn,
                             beforeResolve: (command: ICommand, context: CommandExecutionContext) => void) {
+        // If the command implements the IStoppableCommand interface, we first need to stop the execution of the
+        // previous command with the same action kind and then store the new command as the last stoppable command.
+        if (isStoppableCommand(command)) {
+            const stoppableCommand = this.stoppableCommands.get((command as any).action.kind);
```

Comment:
This is quite dirty: `action` is not defined in the `ICommand` interface, and the property is `protected` in most cases.

Please extract the map key computation to a protected method:
```typescript
protected getStoppableCommandKey(command: IStoppableCommand): string | undefined {
    let key = Object.getPrototypeOf(command).KIND;
    if (key) {
        return key;
    }
    return (command as any).action?.kind;
}
```
If the result is `undefined`, we should skip the code here.

Reply:
Better alternative: extend the IStoppableCommand interface by a public property `keyForStopping` or similar, then the command implementation can decide how to create the key. MoveCommand can just set `keyForStopping = action.kind` in the constructor.